### PR TITLE
Reader: Updates search related analytics.

### PR DESCRIPTION
**Pull request by aerych**
Updates analytics related to reader search.

To test:
Set break points where the relevant stats are bumped.
Test the app and confirm that the stats are bumped as expected and there are no warnings/crashes.

Needs review: @nbradbury if you please sir

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java [hunk 1]
```diff
@@ -221,6 +221,7 @@ public void setArguments(Bundle args) {
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        AnalyticsTracker.track(AnalyticsTracker.Stat.READER_SEARCH_LOADED);
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java]: This fragment isn't used just for search results, to this shouldn't be tracked here.

Comment (aerych) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java]: `onMenuItemActionExpand` is looking like a better option. I'll switch it around.

File: WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java [hunk 2]
```diff
@@ -575,6 +574,7 @@ public boolean onMenuItemClick(MenuItem item) {
             @Override
             public boolean onMenuItemActionExpand(MenuItem item) {
                 resetPostAdapter(ReaderPostListType.SEARCH_RESULTS);
+                AnalyticsTracker.track(AnalyticsTracker.Stat.READER_SEARCH_LOADED);
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java]: 👍

**Comment by nbradbury** [comment]
@aerych This is working great, with one exception: if you rotate the device while the search box is visible, it'll track `READER_SEARCH_LOADED` again. This is because [onMenuItemActionExpand](https://github.com/wordpress-mobile/WordPress-Android/blob/749732aa9bd6a0197af28a73e3f1f9aed0b7586a/WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java#L575) gets called again when the fragment is recreated after rotation.

To avoid this, I recommend only tracking that when search results aren't already showing, like this:
```
public boolean onMenuItemActionExpand(MenuItem item) {
     if (getPostListType() != ReaderPostListType.SEARCH_RESULTS) {
          AnalyticsTracker.track(AnalyticsTracker.Stat.READER_SEARCH_LOADED);
     }
      resetPostAdapter(ReaderPostListType.SEARCH_RESULTS);
     ...
```

**Comment by aerych** [comment]
Thanks @nbradbury.  Ready for one last look.

**Comment by nbradbury** [comment]
Looks good! :shipit:

**Pull request merged**